backstopper-core/src/test/java/com/nike/backstopper/handler/ApiExceptionHandlerBaseTest.java

```diff
@@ -474,6 +475,109 @@ public void doHandleApiException_should_not_allow_error_uid_from_passed_in_extra
                   .isEqualTo(singletonList(result.frameworkRepresentationObj.erv.error_id));
     }
 
+    @Test
+    public void shouldLogStackTrace_should_log_all_errors_when_set_to_StackTraceLoggingBehavior_is_set_to_ForceStackTrace_only_for_an_ApiException() {
+        // given
+        StackTraceLoggingBehavior stackTraceloggingBehavior = StackTraceLoggingBehavior.FORCE_STACK_TRACE;
+        ApiExceptionHandlerBase<TestDTO> handler = new TestApiExceptionHandler();
+
+        List<ApiError> filteredClientErrors = Arrays.<ApiError>asList(new ApiErrorBase("SOME_500_API_ERROR", 99043,
+                "some message", 500), new ApiErrorBase("CUSTOM_400_API_ERROR", 99042, "some message", 400));
+        Exception originalException = new Exception("something erroneous");
+
+        ApiException apiException = ApiException
+                .newBuilder()
+                .withApiErrors(testProjectApiErrors.getGenericServiceError())
+                .withStackTraceLoggingBehaviorAs(stackTraceloggingBehavior)
+                .build();
+
+        // when
+        boolean result = handler.shouldLogStackTrace(500, filteredClientErrors, originalException, apiException, reqMock);
+
+        // then
+        Assertions.assertThat(result).isEqualTo(true);
+    }
+
+    @Test
+    public void shouldLogStackTrace_should_not_log_all_errors_when_StackTraceLoggingBehavior_is_Default_for_an_ApiException() {
```

Comment:
This is not correct. If the StackTraceLoggingBehavior is default or null, then it should fallback to status code checks (i.e. no stack trace on 4xx, full stack trace for everything else).